src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -109,7 +110,7 @@ internal enum SerializationMode
 
     public class XmlSerializer
     {
-        internal static SerializationMode Mode { get; set; } = SerializationMode.ReflectionAsBackup;
+        internal static SerializationMode Mode { get; set; } = RuntimeFeature.IsDynamicCodeSupported ? SerializationMode.ReflectionAsBackup : SerializationMode.ReflectionOnly;
```

Comment:
```suggestion
        internal static SerializationMode Mode => RuntimeFeature.IsDynamicCodeSupported ? SerializationMode.ReflectionAsBackup : SerializationMode.ReflectionOnly;
```
Is it possible to return the constant directly instead of caching it in a static variable, to enable linker to trim the unreachable code?

Reply:
There are some tests checking for the `Mode` setter, e.g. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs#L22-L27

Reply:
Are those even used?  Where are these compilation constants set?
https://github.com/dotnet/runtime/blob/81907137fc454e97a98eb5a7eae21b6c23bcb583/src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs#L21
cc: @StephenMolloy

Reply:
Ah, nevermind, there's a separate project that sets at least one of them:
https://github.com/dotnet/runtime/blob/81907137fc454e97a98eb5a7eae21b6c23bcb583/src/libraries/System.Private.Xml/tests/XmlSerializer/ReflectionOnly/System.Xml.XmlSerializer.ReflectionOnly.Tests.csproj#L3

Reply:
Yeah, at this stage of the game, I'd stick with the original proposed change that caches the value at initialization so we maintain the ability to override when needed. (Which I believe is pretty much just testing right now.) We have a desire to potentially make the reflection-based serializer the primary option in the future, but we've got work to do to bring it up to par with the ILGen serializer before we can do that. Having a switch that is not stuck in one position is helpful.